citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -1113,6 +1114,34 @@ protected void doPostJsonResponse(ParameterParser params,
 		return results;
 	}
 
+	private void validateNestedCollection(List<CitationCollectionOrder> citationCollectionOrders, String citationCollectionId) throws Exception {
+		for (CitationCollectionOrder h1Section : citationCollectionOrders) {
+			if (!h1Section.getSectiontype().equals(CitationCollectionOrder.SectionType.HEADING1)){
+				throw new Exception("The top level sections must all be h1s.");
```

Comment:
Throwing raw Exceptions is bad you should use something more specific. `IllegalArgumentException` maybe?